test/unit/tcache_max.c

```diff
@@ -280,6 +303,7 @@ tcache_check(void *arg) {
 	    "Unexpected.mallctl().failure");
 	expect_zu_eq(old_tcache_max, min_tcache_max,
 	    "Unexpected value for tcache_max");
+	validate_tcache_stack(tcache);
```

Comment:
Reaching this point, the tcache has been rebooted multiple times. Maybe do another `validate_tcache_stack` at the beginning of `tcache_check` as well to make sure it is also using thp when first initialized.

Reply:
Good point. I'll add a few more checkpoints.